src/Symfony/Component/DependencyInjection/Config/AutowireServiceResource.php

```diff
@@ -32,7 +33,21 @@ public function isFresh($timestamp)
             return false;
         }
 
-        return @filemtime($this->filePath) <= $timestamp;
+        // has the file *not* been modified? Definitely fresh
+        if (@filemtime($this->filePath) <= $timestamp) {
+            return true;
+        }
+
+        try {
+            $reflectionClass = new \ReflectionClass($this->class);
+        } catch (\ReflectionException $e) {
+            // the class does not exist anymore!
+            return false;
+        }
+
+        $newSelf = AutowirePass::createResourceForClass($reflectionClass);
+
+        return $this == $newSelf;
```

Comment:
You're right, it's always false, hence the tests would pass anyway. I don't think we need (touch) since my guess is that it handles the case where the original class file does not exist anymore but the same class can be autoloaded from elsewhere (? weird or not I don't no).

In any case we should at least add comments here about what's happening and at best add a test to cover regressions like the two previous PR's.

?